spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java

```diff
@@ -1023,7 +1023,10 @@ public void setAddConversionService(boolean addConversionService) {
 	 * @param defaultProperties the additional properties to set
 	 */
 	public void setDefaultProperties(Map<String, Object> defaultProperties) {
-		this.defaultProperties = defaultProperties;
+		this.defaultProperties = new HashMap<>();
```

Comment:
I think this could be a one liner:

```java
this.defaultProperties = new HashMap<>(defaultProperties);
```

Reply:
Yup -- you're right. Was being too lazy copying from setDefaultProperties(Properties)